addons/mrp/models/mrp_workorder.py

```diff
@@ -400,6 +400,8 @@ def _onchange_date_start(self):
 
     def _calculate_date_finished(self, date_start=False):
         workcenter = self.env.context.get('new_workcenter_id') or self.workcenter_id
+        if not workcenter.resource_calendar_id:
+            raise UserError(_('There is no defined calendar on workcenter %s.', workcenter.name))
```

Comment:
It would be better to raise en error if someone remove the calendar from a workcenter and make the mrp production working without any calendar. It's ok to have overlapping dates. We just don't want to block the mrp user working on the production

Reply:
@reka-odoo yes sure

Reply:
Hello @Whenrow :wave: 
I have added the code.
Please review it :slightly_smiling_face: 
Thanks!

Reply:
@reka-odoo i'm ok with the constraints but to me the production order should be able to be marked as done without issue even if no calendar are set on the workcenters

Reply:
Hello @Whenrow :wave: 
I made the requested changes.
Can you review it? :slightly_smiling_face: 
Thanks!